pkg/manager/impl/validators/partition_validator.go

```diff
@@ -0,0 +1,83 @@
+package validators
+
+import (
+	"fmt"
+
+	"github.com/lyft/datacatalog/pkg/errors"
+	datacatalog "github.com/lyft/datacatalog/protos/gen"
+	"google.golang.org/grpc/codes"
+)
+
+const (
+	partitionKeyName   = "partitionKey"
+	partitionValueName = "partitionValue"
+)
+
+func ValidatePartitions(datasetPartitionKeys []string, artifactPartitions []*datacatalog.Partition) error {
+	if len(datasetPartitionKeys) != len(artifactPartitions) {
+		return errors.NewDataCatalogErrorf(codes.InvalidArgument, "Partition key mismatch, dataset keys: %+v, artifact Partitions: %+v", datasetPartitionKeys, artifactPartitions)
+	}
+
+	// Not all datasets need to be partitioned
+	if len(datasetPartitionKeys) == 0 && len(artifactPartitions) == 0 {
```

Comment:
nit, since you check the length of datasetPartitionKeys and artifactPartitions are equal above you can technically omit the second half of this statement :)

Reply:
good point, done